### PR TITLE
Don't create Doom pools for Wrong name

### LINKED ISSUE #77: Validate Doom Pools exist
No more creating pools when adding doom

### PR DESCRIPTION
**Pull request by SkySom**
Closes #77 

Also adds Create/List/Info commands as well as attempts to find the real name using https://en.wikipedia.org/wiki/Damerau%E2%80%93Levenshtein_distance

### CODE REVIEW

File: src/main/java/logic/DoomLogic.java [hunk 1]
```diff
@@ -71,19 +74,38 @@ public void onEvent(MessageCreateEvent event, Message message, User user, String
...
-                return poolName.equals("") ? DoomHandler.generateDoomEmbed() : DoomHandler.generateDoomEmbed(poolName);
+        String actualPoolName = DoomHandler.findPool(poolName);
+        if (actualPoolName == null) {
+            if (mode == null || mode.isEmpty() || mode.equalsIgnoreCase("list")) {
+                return DoomHandler.generateDoomEmbed();
+            }
+            else {
+                return new EmbedBuilder()
+                        .setTitle("Error")
+                        .setDescription("No Doom Pool with Name ''**" + poolName + "**'' exists.");
```
Comment (Alan19) [inline src/main/java/logic/DoomLogic.java:85]: Can you print suggestions if there are any?

**Review by Alan19 — APPROVED** [review]

File: src/main/java/logic/DoomLogic.java [hunk 2]
```diff
@@ -71,19 +74,38 @@ public void onEvent(MessageCreateEvent event, Message message, User user, String
...
-                return DoomHandler.deletePool(poolName);
-            default:
-                return poolName.equals("") ? DoomHandler.generateDoomEmbed() : DoomHandler.generateDoomEmbed(poolName);
+        String actualPoolName = DoomHandler.findPool(poolName);
+        if (actualPoolName == null) {
+            if (mode == null || mode.isEmpty() || mode.equalsIgnoreCase("list")) {
+                return DoomHandler.generateDoomEmbed();
+            }
+            else {
+                return new EmbedBuilder()
```
[the remark this answers is not in the document]
Comment (SkySom) [reply src/main/java/logic/DoomLogic.java:83]: That'd require a bit bigger of a change here no? None of the current doom stuff does that?

File: src/main/java/logic/DoomLogic.java (hunk 1 above)
Comment (SkySom) [reply src/main/java/logic/DoomLogic.java:85]: I mean for the most part, if it's null it most likely didn't find any within 2 points of similarity , the "multiple" ones is more of an error catch than  an expected.

File: src/main/java/logic/DoomLogic.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Alan19) [reply src/main/java/logic/DoomLogic.java:83]: They don't, but I've been trying to add them when there is an error and doom doesn't throw errors until this gets merged. It's fine if you don't add it here though.

**Pull request merged**